engine/src/lib.rs

```diff
@@ -401,6 +407,71 @@ mod contract {
         // TODO: charge for storage
     }
 
+    /// Callback invoked by exit to NEAR precompile to handle potential
+    /// errors in the exit call.
+    #[no_mangle]
+    pub extern "C" fn refund_on_error() {
```

Comment:
A little bit unfortunate here. We also need to be sure that this can't be abused in anyway. Is it possible to ensure that this can only be invoked by our systems?

Reply:
Literally the next line takes care of this.

```
sdk::assert_private_call();
```